src/microbe_stage/MicrobeAI.cs

```diff
@@ -407,6 +422,40 @@ private void RunAndTumble(Random random)
         }
     }
 
+    /// <summary>
+    ///   Prioritizing compounds that are stored in lesser quantities.
+    ///   If ATP-producing compounds are low (less than half storage capacities),
+    ///   non ATP-related compounds are discarded.
+    /// </summary>
+    /// <returns> A dictionary of absolute priority for each (useful) compound.</returns>
+    private Dictionary<Compound, float> ComputeCompoundsPriorities(List<KeyValuePair<Compound, float>> usefulCompounds)
+    {
+        // Vital compounds are *direct* ATP producers
+        var usefulVitalCompounds = usefulCompounds.Where(x => x.Key == glucose || x.Key == iron);
+
+        // If this microbe lacks vital compounds don't bother with ammonia and phosphorous
+        // This algorithm doesn't try to determine if iron and sulfuric acid is useful to this microbe
+        foreach (var compound in usefulVitalCompounds)
+        {
+            if (microbe.Compounds.GetCompoundAmount(compound.Key) < 0.5f * microbe.Compounds.Capacity)
+            {
+                usefulCompounds = usefulCompounds.Where(x => x.Key != ammonia && x.Key != phosphates).ToList();
+            }
+        }
```

Comment:
Using a .Any statement would simplify this and increase clarity. Roughly:

if ( usefulVitalCompounds.Any(microbe.Compounds.GetCompoundAmount(compound.Key) < 0.5f * microbe.Compounds.Capacity)){
  usefulCompounds = usefulCompounds.Where(x => x.Key != ammonia && x.Key != phosphates).ToList();
}